vhdl_lang/src/config.rs

```diff
@@ -278,6 +283,34 @@ impl Config {
     }
 }
 
+fn substitute_environment_variables(s: &str, replace: &HashMap<String, String>) -> String {
+    let mut result = String::new();
+
+    let mut left = s;
+    while let Some(start) = left.find('$') {
+        // keep non-env var piece as-is
+        result.push_str(&left[..start]);
+        left = &left[start + 1..];
+
+        // replace env var
+        let env_name_len = left
+            .find(|c: char| !(c == '_' || c.is_ascii_alphanumeric()))
+            .unwrap_or(left.len());
```

Comment:
If the string wasn't found, for example because the env is something illegal like `$0abc`, I would prefer this to be communicated to the user.

Reply:
I overlooked that error case, I'll fix it.